### PR TITLE
Feature/2299 api key management UI

### PR DESCRIPTION
**Pull request by amoedoamorim**
## Description

This commit introduces the API Key management UI under Team Integrations page

References: CV2-2299

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Go to team settings > integrations > API Access as a workspace admin and you should be able to create and delete api keys

REQUIRES CHECK API BRANCH https://github.com/meedan/check-api/tree/CV2-4448-api-management-backend while that is not merged

## Things to pay attention to during code review

* Still needs Unit Tests, PropTypes, and meaninful class names to interactive elements
* @brianfleming I'm having trouble figuring how to make the BulletSeparator component wrap and lose the bullets like in the design

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by brianfleming** [comment]
@amoedoamorim I will look at the bulltet thing during review and see what might make sense
one question, is there an API branch that I need to run for this to work locally?

**Comment by amoedoamorim** [comment]
@brianfleming ah yes, I forgot to mention: please use this check-api branch https://github.com/meedan/check-api/tree/CV2-4448-api-management-backend

File: src/app/components/team/TeamIntegrations.js [hunk 1]
```diff
@@ -17,6 +17,22 @@ const TeamIntegrations = () => (<QueryRenderer
         current_team {
           id
           dbid
+          api_keys(first: 10000) {
```
Comment (amoedoamorim) [inline src/app/components/team/TeamIntegrations.js:20]: TODO: Convert to fragment

**Comment by brianfleming** [comment]
> CV2-4448-api-management-backend

@amoedoamorim I put my check-api on that branch and I am getting errors spinning up the app, is that branch fully pushed?

**Comment by amoedoamorim** [comment]
> @amoedoamorim I put my check-api on that branch and I am getting errors spinning up the app, is that branch fully pushed?

let me check that, I think it might be actually missing something

**Comment by amoedoamorim** [comment]
@brianfleming I've pushed something to the check-api branch, please update and rerun the whole docker compose thing to make sure check-api gets restarted and check-web gets the updated relay.json from check-api

File: src/app/components/team/TeamIntegrations.js (hunk 1 above)
Comment (caiosba) [reply src/app/components/team/TeamIntegrations.js:20]: Until here this is the only comment I would add to my review! :D

**Review by caiosba — CHANGES_REQUESTED** [review]
AWESOME work Alex! Until here, the only things I would comment on are the ones you already listed as pending: fragment, PropTypes and unit tests.

**Review by brianfleming — CHANGES_REQUESTED** [review]
@amoedoamorim looking good so far. I will start to put in some design tweaks for you. First off, go ahead and ignore the bullet disappear thing, I don't think its worth the effort.
on to the nit picking though:

- [ ] When I submit the create modal, the underlying page doesn't update automatically, so its confusing, can you have the page react (no pun intended)
- [ ] Can the bullet separator component be updated to have the text be in the correct caption font size for this usage?
- [ ] if I start to submit the create modal, but then cancel it out, then reopen, it retains the previous information, we should clear that out on cancel
- [x] can you vertically center align the textfield and copy and delete buttons?

I will keep poking around, but those are some small changes

**Comment by amoedoamorim** [comment]
Thanks for thte review @brianfleming . I have addressed the last 3 points. The first one should have been working straight out of the box. I've asked @jayjay-w to fix something on the backend and that hopefully will do it.

**Comment by brianfleming** [comment]
Thanks for those changes @amoedoamorim 
let me know when the backend is cleared up request a re-review and I will run through it again, but the changes so far look good

**Comment by amoedoamorim** [comment]
Added unit tests and BEM classes. I think it should be all good from my end but I've asked @jayjay-w to fix the `team.api_keys` sorting, so I'll probably do yet another commit to remove the `apiKeys.splice().reverse()` thing.

File: src/app/components/team/ApiKeyCreate.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+import styles from '../../styles/css/dialog.module.css';
+import inputStyles from '../../styles/css/inputs.module.css';
+import AddIcon from '../../icons/add.svg';
+
+const mutation = graphql`
+  mutation ApiKeyCreateMutation($input: CreateApiKeyInput!) {
+    createApiKey(input: $input) {
+      team {
+        id
+        ...ApiKeys_team
```
Comment (caiosba) [inline src/app/components/team/ApiKeyCreate.js:22]: Nice usage of fragments in side a mutation!

File: src/app/components/team/ApiKeyCreate.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                description="Label for button that creates a new API key"
+              />
+            }
+          />
+        </div>
+      </Dialog>
+    </>
+  );
+};
+
```
Comment (caiosba) [inline src/app/components/team/ApiKeyCreate.js:198]: This is more a personal preference, curious for your thoughts... even when a component doesn't receive any props, I like to declare the empty prop types, just to make it super clear... something like: `ApiKeyCreate.propTypes = {}; // No props`.

File: src/app/components/team/ApiKeyEntry.test.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  expire_at: new Date('2002-02-20').toISOString(),
+  user: {
+    name: 'Daddy Shark',
+  },
+};
+
+const activeKey = {
+  id: '123xyz',
+  title: 'Test API Key',
+  access_token: 'babysharkdododoo',
```
Comment (caiosba) [inline src/app/components/team/ApiKeyEntry.test.js:20]: :baby: :shark:

File: src/app/components/team/TeamBots.js
Comment (caiosba) [inline src/app/components/team/TeamBots.js:1]: Awesome, exactly the kind of minimal intervention we discussed.

**Review by caiosba — APPROVED** [review]
Awesome work, @amoedoamorim !

**Comment by caiosba** [comment]
I tested it locally and it works fine. I just noticed one minor thing: the "Connect with us directly for support" link does nothing... shouldn't we at least make it a `mailto:` link to the `SUPPORT_EMAIL`?

**Comment by amoedoamorim** [comment]
> I tested it locally and it works fine. I just noticed one minor thing: the "Connect with us directly for support" link does nothing... shouldn't we at least make it a `mailto:` link to the `SUPPORT_EMAIL`?

I suppose you don't have the intercom token set in your local config, then intercom won't pop up.

File: src/app/components/team/ApiKeyCreate.js (hunk 3 above)
Comment (amoedoamorim) [reply src/app/components/team/ApiKeyCreate.js:198]: Yeah, this is better. Will do!

**Comment by caiosba** [comment]
> > I tested it locally and it works fine. I just noticed one minor thing: the "Connect with us directly for support" link does nothing... shouldn't we at least make it a `mailto:` link to the `SUPPORT_EMAIL`?
> 
> I suppose you don't have the intercom token set in your local config, then intercom won't pop up. Do you mean we should leave the mailto: as a fallback? If Intercom is not defined then e.preventDefault doesn't happen and it works as an email link instead?

Ah, OK, yeah, I don't have Intercom configured locally... I think we can leave it as is for now but maybe just add a comment to the code explaining this.

**Review by brianfleming — APPROVED** [review]

**Pull request merged**